runtime/lua/vim/shared.lua

```diff
@@ -267,18 +267,23 @@ function vim.tbl_deep_extend(behavior, ...)
 end
 
 --- Deep compare values for equality
+---
+--- Tables are compared recursively unless they both provide the `eq` methamethod.
```

Comment:
weird that we named this `deep_equal` instead of `eq_something` for parallel with Lua's convention